### DISCUSSION TITLE
Are hooks not triggered if the record is created inside a custom route?

### DISCUSSION THREAD

**Discussion by shubhendumadhukar** [pocketbase/pocketbase#2045] [Q&A] [answered]
I have a custom route which creates a record using `form.Submit()` as shown [here](https://pocketbase.io/docs/record-methods/#create-new-record-with-data-validations), which is working correctly, i.e. I am able to create the record. However, I am expecting an `OnRecordAfterCreateRequest` to be triggered after the record is created, which doesn't happen. Note that the `OnRecordAfterCreateRequest` works fine if I create the record from my frontend.

Is this an intended behaviour? If yes, is there a way I can get the record/id of the record created by `form.Submit()`?

**Comment by ganigeorgiev** [comment, marked as answer]
The `*Request` hooks are triggered only on an API request.

We can't trigger the `OnRecordAfterCreateRequest` hook on programmatically inserted records because there is no HTTP context (aka. the request) associated with the action.

If you want to listen to record create events, you can attach to the the [`OnModel*` db/dao hooks](https://pocketbase.io/docs/event-hooks/#db-hooks).

**Reply by shubhendumadhukar** [reply]
Thank you! That worked.

I think I know the answer to this, but just to confirm, `OnModel*` hooks will be triggered on both Model events as well as Request related events?

**Reply by ganigeorgiev** [reply]
> OnModel* hooks will be triggered on both Model events as well as Request related events?

Yes. `OnModel*` are triggered on each create/update/delete DB operation that was performed via the default `app.Dao()`. A single HTTP request could fire one or many `OnModel*` hook events.
